Update base-pt-PT.yaml

ninja-spotted opened the pull request:
Small revision of the translation and corrections

ozgurozbek committed:
Update base-tr.yaml

translated all remaining English parts
fix typos
better convey meaning in some sentences

EmeraldBlock committed:
ES modules and config refactor

- switches to ES modules in gulp and src
- uses dengr's webpack configs and package.json, with modifications
- removes gulp/package.json
- removes babel stuff
- removes gulp-load-plugins, instead importing gulp plugins manually
- removes unused and trivial packages
- upgrades packages
- uses path/posix, for gulp
- removes __dirname in favor of relative urls

DJ1TJOO committed:
Removed china, wegame, gog and demo (#3)

* Removed build and global variables for china, wegame
* Removed google analitycs

Bagel03 committed:
Add ts support

Bagel03 committed:
Prettier

EmeraldBlock committed:
build fixes

DJ1TJOO committed:
Fixed missed trim replacement (#9)

Bagel03 committed:
Update ts to 5.0 (#11)

dengr1065 committed:
Remove GOG/Wegame source roots

Bagel03 committed:
Add support for JSX

Allows usage of JSX across the codebase, JSX is turned into regular HTML elements.

dengr1065 committed:
Clean up and update workspace files

* Remove deprecated TSLint config

* Remove jsconfig.json

jsconfig.json is an alternative name for tsconfig.json that has allowJs
set to true. Since the directory already contains a tsconfig.json file,
this file does nothing.

* Remove Gitpod configuration

Gitpod is not as useful in CE because we're focusing on standalone.
Also, the configuration was added just to promote the service and there
are other similar services which may be preferred by the user.

* Remove Travis CI configuration

Travis CI is not used in shapez anymore in favor of GitHub Actions. In
addition to that, CI setup for the community edition will be done once
the codebase is in a better condition.

* Remove CONTRIBUTORS and CONTRIBUTING.md

The contributors file is out of date and unneeded as GitHub now shows
all collaborators anyway - however it could be still used in the future
to show a list of contributors in the game. CONTRIBUTING.md contains the
text of CLA, which is not enforced in this repository.

* Replace VSC workspaces with regular settings files

VSCode workspaces don't provide much benefit in a source tree like
shapez. This commit also adds EditorConfig and ESLint VSCode extensions
to the recommended extensions list.

* Remove CI workflow for building the game

It hasn't been updated in a while and we don't need it for now. Once the
codebase is in a more managable state, a new workflow will be added.

* Remove some unused resources

There are still many resources left, but they will be removed along with
the relevant code later.

edorfaus committed:
Fix build by re-adding a missing dev dependency

This fixes cssnano being unable to load the preset "advanced", after
the dependency for that was removed in commit 24ceb6664d.

ninja-spotted committed:
Update base-pt-PT.yaml

Small revision of the translation and corrections

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/tobspr-games/shapez.io?pullRequest=1578) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/tobspr-games/shapez.io?pullRequest=1578) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/tobspr-games/shapez.io?pullRequest=1578) it.</sub>

Bagel03 committed:
Add Most Useful TS files (#13)

* Update Signal

* Update modal_dialogs

* Inputs

* Update factories

* Update tracked state

* Changed let to const where possible

* Add HUD typings

* improvements to typings

* fix exports not being exposed to mods

* fix signal typings

* remove TypedSignal

* fix all reported type errors

---------

Co-authored-by: Thomas B <t.ferb1@gmail.com>
Co-authored-by: EmeraldBlock <yygengjunior@gmail.com>

DJ1TJOO committed:
Added information about community edition (#2)

* Added information about community edition

* Update README.md

* README rewrite

---------

Co-authored-by: EmeraldBlock <yygengjunior@gmail.com>

dengr1065 committed:
Remove the restriction manager

Restriction manager is completely removed as it's only useful for demo
version. Lots of other demo-related things are left, such as ad support
or extended telemetry. This should be cleaned up later.

dengr1065 committed:
Slight improvements for the build process (#19)

* Unify the version source

Changes the development config to use version from package.json and
makes the production config use the same utility function. Also removes
the version file as it's no longer needed.

* Remove some Steam references from build process

Removes files needed to publish the game on Steam and related tasks that
can be only used by the developer anyway. Only the build process is
affected, the actual game part still supports Steam integration.

* Refactor/add packaging tasks

Adds new Gulp tasks to create a distributable package of the game. A
task for macOS is not provided because signing needs to be figured out
first. Package creation tasks are also aliased in package.json for
future usage in external tools or CI. Aside from that, alternative
methods of downloading the libGDX Texture Packer are dropped.

dengr1065 committed:
Fix missing Gulp CWD in packaging scripts

The gulpfile is in a subdirectory and the scripts need to take that into
account. While there's already one such script, I forgot to add the
arguments in the new ones.

dengr1065 committed:
Remove ads, analytics and Steam SSO, simplify HTML tasks (#21)

* Remove ad support, analytics and Wegame leftovers

The game may be somewhat broken in a few places, but it doesn't matter
for now. This is still not the end.

* Remove Steam SSO and demo stuff

Steam SSO is completely removed, a few things from demo like simplified
level sets are gone as well. Puzzle DLC on the other hand is now always
"owned" and will ask for a token to log in.

Removes

* Use shapez dialogs for Puzzle DLC token input

Yes, this sucks *a lot*. But it's a temporary measure, trust me :P

* Simplify HTML tasks

Removes the web (demo) index.html page and makes HTML tasks independent
of the build variant. This might not be the best solution, but it works
for now.

dengr1065 committed:
Add more packaging tasks and fix localConfig not being created (#22)

* Update Electron dependencies

This will break some mods and a few things were deprecated, but the
Electron wrapper will be replaced anyway.

* Add macOS and arm64 packaging tasks

As @chunkybanana has tested the general approach to building shapez on
macOS, I can now add these tasks. Aside from that, now there are tasks
to build for the 64-bit ARM variant of each system and a task that
builds packages for all platforms and architectures at once. A bug where
localConfig wasn't created when building for the first time was also
fixed.

dengr1065 committed:
Remove cache busting

Cache busting is not required in standalone context.

EmeraldBlock committed:
be more consistent with drawing methods

EmeraldBlock committed:
make tiled sprites pixel-aligned to avoid transparent seams

EmeraldBlock committed:
remember about blueprints

EmeraldBlock committed:
fix hub double-rendering

EmeraldBlock committed:
remove mod_examples

EmeraldBlock committed:
Squashed commit of the following:

commit 176343785eea110e529f26027bf864ae04068384
Author: EmeraldBlock <yygengjunior@gmail.com>
Date:   Fri Dec 8 23:03:51 2023 -0600

    update readme

commit 8c1c3a0c47f5125126cb00d32a48f4f9344a3fb3
Author: EmeraldBlock <yygengjunior@gmail.com>
Date:   Fri Dec 8 23:00:05 2023 -0600

    fix bugs

commit ea881e68c693a447e0698a3a6e7cfb1f25ccb6cc
Author: EmeraldBlock <yygengjunior@gmail.com>
Date:   Fri Dec 8 22:46:46 2023 -0600

    expose all tasks with old api

commit fa6d7a3920ff573eadb61425cc077f0e00406164
Author: EmeraldBlock <yygengjunior@gmail.com>
Date:   Fri Dec 8 21:51:20 2023 -0600

    switch to exported gulp tasks

commit 348b19a0171e65400bcd434cf7b7432f3488a411
Author: EmeraldBlock <yygengjunior@gmail.com>
Date:   Mon Nov 20 22:55:38 2023 -0600

    parallelize dev build

commit 56de73e2d18d20e5ea7202afc021573a746e5012
Author: EmeraldBlock <yygengjunior@gmail.com>
Date:   Mon Nov 20 20:44:10 2023 -0600

    use promises in gulpfile

commit 6ab54372482f26acb4769428eefbdc48240a12a1
Author: EmeraldBlock <yygengjunior@gmail.com>
Date:   Mon Nov 20 20:33:36 2023 -0600

    make java -version print again

commit b0e4cf57bdc404bb3b0e45b7b233d5f7648c800e
Author: EmeraldBlock <yygengjunior@gmail.com>
Date:   Mon Nov 20 20:14:13 2023 -0600

    use promises for gulp tasks

EmeraldBlock committed:
improve dev watching

dengr1065 committed:
Merge pull request #27 from tobspr-games/refactor/gulp

Refactor gulp setup; switch from webpack watch to gulp.watch exclusively

dengr1065 committed:
Redo ESLint and TypeScript compiler config

Many configuration files in this repository were created a long time
ago, then were modified as problems occurred. Now that there is
TypeScript support, it makes sense to clean up this mess, at least by
making small steps. This configuration is based on strict settings, but
most of these are currently disabled - otherwise it would be too hard to
work with existing JavaScript code. The downside of this change is
pollution of files with warnings and errors, even though they are valid.

- ESLint/TypeScript upgraded
- TS configuration is now shared between arbitrary Node scripts, Gulp
  files and the Electron wrapper
- A few eslint-disable comments are removed

dengr1065 committed:
Merge pull request #26 from tobspr-games/chore/remove-mod-examples

Remove mod examples

dengr1065 committed:
Simplify ESLint file matching patterns

dengr1065 committed:
Merge pull request #30 from tobspr-games/dengr1065/eslint-tsc-revamp

Redo ESLint and TypeScript configuration

dengr1065 committed:
Modify ESLint configuration

Fix minimatch patterns, disable no-unused-vars entirely, stop calling
tsc in the lint script.

Commit:
Remove all references of "G_IS_STANDALONE"

Commit:
Remove folders and steam stuff

codeman4033 committed:
Fix import assertion

dengr1065 committed:
Merge pull request #32 from cody-ferguson/feat/remove-all-web-stuff

Remove all web related stuff

dengr1065 committed:
Update Prettier to 3.3.2

dengr1065 committed:
Update formatting configurations

EditorConfig applies to all files, LF is enforced instead of CRLF,
Prettier configuration moved to JSON, properties are inherited from
EditorConfig.

dengr1065 committed:
Merge .gitignore files

Prettier only takes the root .gitignore into account, which leads to
formatting of auto-generated files. Moving ignore patterns into the root
.gitignore file makes Prettier follow them and cleans up the workspace.

dengr1065 committed:
Apply formatting to src/js

dengr1065 committed:
Apply formatting to src/css

dengr1065 committed:
Apply formatting to gulp

dengr1065 committed:
Apply formatting to the rest of files

dengr1065 committed:
Update prettier-all script

Also exclude files with extensions that are usually large. Reduces the
format time by approximately 4 seconds.

dengr1065 committed:
Merge pull request #31 from tobspr-games/dengr1065/eslint-config2

Fix and modify ESLint configuration

dengr1065 committed:
Merge branch 'master' into dengr1065/formatting-update

dengr1065 committed:
Merge pull request #33 from tobspr-games/dengr1065/formatting-update

Update formatting configurations and format all files

dengr1065 committed:
Update indirect dependency fluent-ffmpeg

fluent-ffmpeg@2.1.3 fixes support for FFmpeg 7.x.

dengr1065 committed:
Merge pull request #34 from tobspr-games/dengr1065/fix/update-fluent-ffmpeg

Update fluent-ffmpeg to 2.1.3

dengr1065 committed:
Remove unused webpack files

dengr1065 committed:
Remove some mobile-web code

dengr1065 committed:
Get rid of more Steam leftovers

dengr1065 committed:
Remove more unused styles

dengr1065 committed:
Fix external link handling

dengr1065 committed:
Remove FTP tasks

dengr1065 committed:
Remove unused typehints block

dengr1065 committed:
Merge pull request #35 from tobspr-games/dengr1065/more-cleanups

Various cleanups

dengr1065 committed:
Fix shapez exports not being exposed in production

A previous commit has mistakenly changed the behavior of a condition
checking for standalone version. Instead of always being true, shapez
exports were only exposed if the game was built in development mode.

dengr1065 committed:
Merge pull request #36 from tobspr-games/fix/shapez-exports-nondev

Fix shapez exports not being exposed in production

dengr1065 committed:
Avoid direct HubGoals#storedShapes access

Use getShapesStored and getShapesStoredByKey when possible instead. Only
covers usages outside of hub goals.

dengr1065 committed:
[TS] Rename TextualGameState

dengr1065 committed:
Resolve @/jsx-runtime in webpack

TypeScript compiler on its own will only inject this import, but not
resolve it. The bundler is supposed to resolve the import, so just do
that.

Dani20000 committed:
Update italian translation (#38)

dengr1065 committed:
[TS] Rename GameState

dengr1065 committed:
Allow Node in JSX runtime

The typings are not consistent, but they seem to work. I'll leave the
rest as-is and just hope that someone fixes them later.

dengr1065 committed:
Use @ exclusively for root imports

Remove "root" resolve alias and now-redundant jsx-runtime alias.

dengr1065 committed:
[TS] Rename TextualGameState

This time with .tsx extension.

dengr1065 committed:
Fix all instances of "reciever" typo

dengr1065 committed:
Basic JSX support in states

Quite hacky considering the complex indirection and minor differences in
the appearance of each state, as well as legacy code and need to support
HTML strings for now. Some methods could be improved, refactored or
deleted, but no major changes were made.

Mods and puzzles menu states are broken in this commit.

dengr1065 committed:
[TS] Rename ModsState

dengr1065 committed:
Rewrite mods state

Most existing mods aren't going to work in CE and the system will be
reworked some time later, so this will work for now.

dengr1065 committed:
Merge pull request #37 from tobspr-games/dengr1065/avoid-storedshapes-direct-access

Replace some outside usages of HubGoals#storedShapes

dengr1065 committed:
Remove accessNestedPropertyReverse

Obfuscation is useless in an open source game.

dengr1065 committed:
Remove some legacy browser support code

Includes prefixed document.hidden property with respective
visibilitychanged event and redundant support for navigator.language /
lack of navigator.languages check.

dengr1065 committed:
Remove compression prefix fallback

All modern browsers should support \1 as the compression prefix, even
outside of Electron or recent Chromium versions. Local storage /
IndexedDB platform wrappers were removed a while ago.

dengr1065 committed:
Minor globalConfig and unused code cleanup

Remove secrets that are no longer used and fetchDiscounts preload step.

dengr1065 committed:
[TS] Rename globalConfig

dengr1065 committed:
Remove prefixed imageSmoothingEnabled

webkitImageSmoothingEnabled got deprecated back in 2016.
globalConfig.smoothing.quality is now properly typed as well.

dengr1065 committed:
Remove some polyfills

The Math polyfills might still be needed (even though they are polluting
the Math object) and I'm not sure about DOM polyfills yet.

dengr1065 committed:
Drop support for SHA1 checksums

Also remove an unused function and switch to crypto.randomUUID() for
savegame ID generation.

dengr1065 committed:
Fix Vector#equalsEpsilon ignoring parameter

Provided epsilon parameter was used for Y axis but ignored for X axis.
Some useless return statements are also removed.

dengr1065 committed:
Merge pull request #40 from tobspr-games/dengr1065/even-more-cleanups

Even more cleanups

dengr1065 committed:
Update Electron

dengr1065 committed:
Modify Electron wrapper

Minimal set of changes, not a refactor yet. Make shapez store all of
its data in a single directory whose path is provided by Electron APIs.
Useless checks and web preferences are removed and app.isPackaged is
used to detect whether the app should try to load a localhost URL.

dengr1065 committed:
Merge pull request #41 from tobspr-games/dengr1065/electron-updates

Update the Electron wrapper

dengr1065 committed:
Remove unused build variants

Only remove a small amount of code, other changes to build variants are
pending.

dengr1065 committed:
Remove changelog skins

The feature wasn't used in the last 2 years anyway.

dengr1065 committed:
Clean up SCSS stylesheets

Can have minor effect on game appearance, most importantly the removal
of letter-spacing by default. Styles are now much lighter though.
Changes all keybinding code to use kbd instead of code as well.

dengr1065 committed:
Remove some unused resources

There might be more that I haven't noticed yet.

Ecconia committed:
Fix runnable-textpacker.jar download

Adds 'await' to ensure the download will be complete.

Without this the download might never complete. This issue occurred on a Linux OS (which seem to handle file streams a bit different than some other OSs).
Effects of not fixing this are that the java execution following this download fails for funny "corrupted jar" and other reasons - depending on the size and content of the incomplete jar file.

dengr1065 committed:
Merge pull request #42 from Ecconia/fix-texturepacker-download

Fix runnable-textpacker.jar download

dengr1065 committed:
Merge pull request #43 from ozgurozbek/patch-1

Update base-tr.yaml

dengr1065 committed:
Simplify GameState legacy HTML string support

Move the getInnerHTML fallback to the default implementation of
getContentLayout; this allows for code reuse in TextualGameState too.

dengr1065 committed:
Merge pull request #39 from tobspr-games/dengr1065/textual-game-state-work

Basic support for JSX/TSX in GameStates

dengr1065 committed:
Fix TextualGameState HTML string fallback

Move the additional wrapping for getMainContentHTML into getInnerHTML
and call it from GameState super method. Also apply the same wrapping to
getInitialContent (if not null) to ensure equal treatment of legacy HTML
strings and modern JSX/TSX methods.

dengr1065 committed:
Merge pull request #44 from tobspr-games/fix/tgs-fallback

Fix TextualGameState HTML string fallback

MohamedAAbdallah committed:
Adding full Arabic Translation (#46)

* Updated steamPage AR Translation

* Updated global AR Translation

* Updated demoBanners AR Translation

* Updated mainMenu AR Translation

* Updated dialogs AR Translation

* Updated ingame AR Translation

* Updated shopUpgrades AR Translation

* Updated buildings AR Translation

* Updated storyRewards AR Translation

* Updated Settings AR Translation

* Updated keybindings AR Translation

* Updated about AR Translation

* Updated changelog AR Translation

* Updated demo AR Translation

* Updated tips AR Translation

* Updated puzzleMenu AR Translation

* Updated backendErrors AR Translation

* Updated game name to be more cosistent

* Updated dialogs to be less than the 200 char limit

* Rewrote the storyRewards to be less than the 200 char limit

* Fixed syntax errors

dengr1065 committed:
Rewrite the Electron wrapper (#47)

* Migrate Electron wrapper to ESM

Use ESM syntax for main process, move some fs usages to fs.promises,
switch to import.meta.url/import.meta.dirname to handle file paths;
clean up redundant code.

* Add TypeScript support to Electron wrapper

The support is very basic, tsc is used to transpile code. Build scripts
are modified to not copy any Electron code other than preload.cjs and
use an extremely cursed setup to call the TypeScript compiler.

* [TS] Rename platform/storage

* Rewrite Electron wrapper

MVP, missing some features from the old wrapper and most planned
features. Some of the functionality hasn't been verified.

dengr1065 committed:
Add missing TypeScript package and update readme

Unfortunately we have to keep a second copy for now. The lockfile
specifies the same version as used in the root package to avoid issues.
This may be replaced with Node.js type stripping in the future.

README.md is updated to include the correct command for testing the
Electron wrapper.

dengr1065 committed:
Fix file writes to non-existent directories

When the game attempts to create the settings file but its directory
does not exist yet, an early error occurs. In reality, there is no way
for the directory to be created without user intervention. This change
makes sure the directory is created before any attempt to write a file
is done. Errors related to directory creation and/or file writes are
still reported as usual.

dengr1065 committed:
Handle window.open calls with external browser

Re-use the same openExternalUrl logic meant for frame navigation to
handle window.open calls as well. Possibly a partial solution, as the
ability to open new windows might be useful for mods.

dengr1065 committed:
Update TypeScript packages

Yarn seems to complain due to my hand-edited package.json, but
TypeScript packages are out of sync if I let it update the electron one.
To avoid these issues, update both packages.

dengr1065 committed:
Mass-rename rotater -> rotator (#51)

Rename all references of rotater to rotator, including code, styles,
assets and translation files.

dengr1065 committed:
Remove theme-short.mp3

Not used anywhere. Reduces size of each uncompressed bundle by 6 MB.

dengr1065 committed:
Merge pull request #55 from tobspr-games/dengr1065/remove-short-theme

Remove unused theme-short.mp3

dengr1065 committed:
Merge pull request #49 from tobspr-games/dengr1065/electron-fixup

Fix some issues with the Electron wrapper refactor

dengr1065 committed:
Check fullscreen state before modifying it

Workaround for an Electron bug which causes menus to disappear when
using BrowserWindow#setFullscreen(false).

dengr1065 committed:
Merge pull request #59 from tobspr-games/dengr1065/fullscreen-menu-bug

Fix disappearing window menu on Linux

dengr1065 committed:
Remove most of "preloader" code (#60)

Make the build process less complicated and reduce scatter
of renderer source code.

dengr1065 committed:
Preliminary ASAR mods support in main process

Consider mods broken in this commit. Many areas are unfinished and use
weird or missing types.

dengr1065 committed:
Remove leftover callback check

Was removed completely along with the preloader script.

dengr1065 committed:
Re-use global app variable in ModLoader

The linkApp method is removed as it was unused. A getter is used to
ensure best development experience. It can be further improved by adding
a guard for cases where GLOBAL_APP is not set yet.

dengr1065 committed:
Basic ignore of duplicate mods

Only load the mod with highest priority for now. Proper behavior would
be to keep metadata for all duplicates and load the first that isn't
disabled. This allows easy mod source switching (for example, user mods
and development mods)

dengr1065 committed:
Remove externalModUrl

Remove the debug option and code that used it. It is redundant due to
existence of --load-mod in the Electron wrapper.

dengr1065 committed:
Remove game version check

Doesn't make much sense with how experimental CE is right now. A
different logic for checking whether a mod is compatible will be used
later.

dengr1065 committed:
Remove unused build pipeline code

dengr1065 committed:
Remove achievement support

Suggested by @SkimnerPhi. If needed, mods should provide their own
support for achievements. For vanilla this is dead code.

dengr1065 committed:
Remove savegame stats (achievements data)

No longer used as achievements are gone.

Also, avoid copying the old savegame dump object. This is possible
thanks to this change.

dengr1065 committed:
Merge pull request #63 from tobspr-games/dengr1065/remove-achievements

Remove support for achievements

dengr1065 committed:
Remove some unused translation strings

Keep some that are either still used but shouldn't be or those that may
be used in the future. Files other than base-en.yaml are first synced
with the sync-translations.js script, then formatted with prettier-all.

dengr1065 committed:
Migrate from Yarn Classic to npm (#65)

* Switch to npm in electron/

May change resolved package versions.

* Remove redundant electron-notarize dependency

The number of dependencies stays the same. Only direct mention in
package.json is removed.

* Assume Node.js 22.x and clean up dependencies

Targeting Node.js 22.x allows using the new fs.glob* functions, which
can replace the glob module. It is still downloaded as there are other
packages using it, but is no longer included in devDependencies.

Also remove @types/filesystem, as this API is not used anywhere in the
code, it is non-standard and some of it was removed from Chromium. Was
likely used with Cordova for YORG.io 3 mobile support.

* Update linting stack

Install the latest versions of the following packages:

- @eslint/js (^9.24.0)
- eslint (^9.24.0)
- typescript-eslint (^8.29.1)

and remove packages that are no longer used:

- @types/eslint__js
- yarn

Config files were not modified.

* Switch root package to npm

npm is able to resolve packages using yarn.lock as a resolution hint.
Note that files other than those included in this commit are
intentionally kept outdated, as they may need a bigger change.

dengr1065 committed:
WIP: Basic ASAR modding in renderer

Also fix a few issues in Electron code. This is not as polished yet, UI
from old mod support was reused for now and is likely broken. Mods can
be loaded, but there isn't much QoL around the support for now.

dengr1065 committed:
Respect --safe-mode once again

The new mod locator system now respects the --safe-mode command line
switch, ignoring the user and distro mod directories.

dengr1065 committed:
Create ModInterfaceV2 for cool stuff

For now it only includes a helper function to resolve file URLs and a
method that adds a CSS stylesheet to the document.

dengr1065 committed:
Fix broken mods state appearance

Not a complete rework. This is just a temporary improvement.

dengr1065 committed:
Fix savegame load error on missing mod website

Not a proper fix, but this thing needs a huge refactor anyway. Goodbye,
a bug that existed ever since the first version of mod support!

dengr1065 committed:
Remove some outdated browser checks

Also remove some type definitions that are definitely no longer needed
or were never correct.

dengr1065 committed:
Remove mentions of ads and twitter

:put_litter_in_its_place:

dengr1065 committed:
Replace Object#hasOwnProperty with Object.hasOwn

Change most, but not all usages. This is mostly to please the linter,
but Object.hasOwn is also slightly shorter, especially when compared to
the proper form, Object.prototype.hasOwnProperty...

dengr1065 committed:
Remove non-standard attribute assignment

Don't set the non-existent "opaque", "webkitOpaque" and "mozOpaque"
properties on a canvas. Keep using { alpha: false } when initializing
the context, which is the correct way.

dengr1065 committed:
Remove savegame object compression

Preparation for further changes. Savegames that are not compressed are
much bigger, but this is a rather complex algorithm that shouldn't
exist.

dengr1065 committed:
Pass storage object to ReadWriteProxy

Make it slightly easier to replace the storage interface used for
app_settings.bin and savegames.bin in case it's ever needed. Savegames
always use app.storage for now, but this should be easier to change as
well.

dengr1065 committed:
Add dedicated saves storage, new fs job types

Keep track of the storage ID in each renderer Storage instance and pass
it to the IPC bridge. Jobs are dispatched to the relevant handler (only
saves/ for now) and all (de)compression is handled there.

Add dedicated fs-job types to read or write and (de)compress data
from/to the file picked by the user. Remove redundant utility functions
that used web APIs instead.

dengr1065 committed:
Enable Arabic language

As per author's request. The translation was already good enough, and
now the author wrote to me about this.

dengr1065 committed:
Remove some compression leftovers

Remove the CRC dependency and "binary file salt" in globalConfig.

dengr1065 committed:
Merge pull request #69 from tobspr-games/dengr1065/new-savegame-storage

Implement gzipped MessagePack savegame storage

dengr1065 committed:
[TS] Rename EntityManager

dengr1065 committed:
Use Map and Set for entity storage

This is not a big optimization but an optimization nonetheless. Mostly
based on awesome work by @Xiving. Further work should be done to get
most out of these changes.

dengr1065 committed:
Use Set for entities access during belt recompute

Create a method to return all entities with the specified component as a
set (without array indirection), and use the method for belt
recomputations.

dengr1065 committed:
Avoid null uid in EntityManager#registerEntity

Apply suggestion by @chunkybanana to improve readability.

dengr1065 committed:
Fix unhandled savegame error not being displayed

Move the file loading code into the try/catch block, as it can and will
produce errors eventually.

dengr1065 committed:
Remove main process compression

To be simplified and replaced with renderer-side web workers.

dengr1065 committed:
Clean up Electron wrapper code

Remove a temporary console.log statement as well as now-unused msgpack
module.

dengr1065 committed:
Fix main menu not waiting for savegame import

Does not fix the savegame list bugs. Note that savegame index
serialization happens a few times for some reason, this is out of scope
for this commit though.

dengr1065 committed:
Re-implement compression workers and abstraction

Implement DefaultCompression class along with a generic interface to
facilitate easy to use compression in a background thread, and make use
of this class in Storage implementation by default.

dengr1065 committed:
Compression: Terminate workers after use

While the browser engine can clean up workers automatically, it's a good
idea to terminate them as soon as we don't need those anymore. It's
generally suggested to reuse a single worker for repetitive operations,
but implementing such a system is much harder. In addition, spawning a
new worker for each operation allows (de)compressing multiple blobs in
parallel :)

dengr1065 committed:
Basic FS watcher for --load-mod mods

Add a --watch command line flag to be used in tandem with --load-mod.
Chokidar is used to detect file updates, and a page reload is triggered
when there are any changes to watched mod files. Only mods loaded via
--load-mod are affected.

This implementation has a minor issue with how cache is cleared -
removing disk cache is a bit too aggressive - but the only alternative I
could find is to use a non-persistent Electron session, which would get
rid of disk cache entirely (this is not a concern).

Commit:
Added gameInitialized check to onEntityChanged in map_view.js

Sylviromi committed:
Merge pull request #75 from tobspr-games/tceprepk/map-view-optimization

Added gameInitialized check to onEntityChanged in map_view.js

dengr1065 committed:
Remove DOM remove() polyfills

Remove polyfill functions for Element, CharacterData and DocumentType
classes, as these methods are all Baseline Widely Available. Also get
rid of redundant IIFE for the Math "polyfills".

dengr1065 committed:
Remove beginRoundedRect utility

No longer needed, as CanvasRenderingContext2D#roundRect is a built-in
way to achieve the same result now.

dengr1065 committed:
Remove redundant/unused stuff from globals.d.ts

Remove things that no longer exist or are covered by built-in TypeScript
libraries. Also switch to webpack-provided types instead of defining
them manually.

dengr1065 committed:
Move beginCircle to polyfills.ts

Technically not a polyfill, but neither are two existing functions in
polyfills.ts.

dengr1065 committed:
Remove unused G_HAVE_ASSERT define

Not used anywhere except exposing BUILD_OPTIONS for mods.

dengr1065 committed:
Ensure proper code splitting for translations

Make updateApplicationLanguage load translations asynchronously as
separate chunks in a slightly hacky way. Also implement a workaround for
old translation overrides registry ModInterface API.

dengr1065 committed:
Do not compress translation files

Remove the compressjson loader and related defines in webpack config.
Also remove the remaining lz-string usages in the project and a local
copy.

dengr1065 committed:
Inline {enable,disable}ImageSmoothing usages

The method isn't as complex as it used to be, and is only used in two
places. It is simpler to have it inlined instead.

dengr1065 committed:
Remove highly redundant utils functions

Remove isSupportedBrowser entirely and inline usages of getLogoSprite.

dengr1065 committed:
Remove unused Twitter link stuff

No longer exists, lmao.

dengr1065 committed:
Manually debounce force reload event

Remove chokidar built-in event debouncing/aggregating and replace it
with a basic custom implementation with a 250ms settle delay.

dengr1065 committed:
Tidy up ALEA RNG code

Not a performance improvement, just something that needed to be done.
Slightly simplifies the structure and removes useless functions
(changing the seed resets the RNG anyway - it is better to create a new
instance instead).

dengr1065 committed:
Build tools cleanup pass (#81)

* Fix tsconfig scopes affecting html.js

Since it's quite hard to use a DOM library type there, remove the type
entirely.

* Remove environment variables check

Nothing is using them anymore. It can be added back if needed later.

* Refactor Texture Packer downloading

Refactor local-config.js tasks file into a generic "environment"
category consisting of checking if Java is installed, downloading the
runnable Texture Packer if it's not yet downloaded and copying the local
configuration template; update README accordingly.

* Prepare environment only at postinstall

Remove environment.prepare task from default build pipelines, add a
postinstall script that calls the task, using environment.js as the
gulpfile to speed it up.

* Remove "docs" tasks and types generation script

Remove tasks from docs.js as they are unlikely to do anything meaningful
nowadays. Also remove the buildTypes script as it doesn't work anymore.
A better solution will be provided in the future.

* Simplify some globs

Use additional gulp.src options instead of specifying more or complex
globs.

* Extract built-temp location to a variable

Add the src/js/built-temp directory as a new variable in config.js,
replace all existing references to built-temp with this variable.

dengr1065 committed:
Do not automatically open devtools

Remove the behavior of automatically opening developer tools when
launched with --dev command line switch. As --hide-devtools switch is
now unused, remove it as well.

dengr1065 committed:
Minor "buffers" cleanup

Remove the canvas context loss handling as it was incorrect, simplify
the error conditions in makeOffscreenBuffer and remove commented out
code.

dengr1065 committed:
Rename incorrectly named .d.ts files

These files do not describe existing JS types and instead define their
own. Usage of .d.ts extension prevents these types from being emitted
when using emitDeclarationOnly, which affects generation of typings for
mod development.

dengr1065 committed:
"Fix" ModInterface not accepting mod:// URLs

Remove the check of URL beginning with data:image. This is an old API,
so leave other quirks as-is.

dengr1065 committed:
Remove unused config properties

Remove globalConfig properties that are not used anywhere, simplify
mobile device detection (mostly unused for normal builds) and merge or
inline config constants where it makes sense.

dengr1065 committed:
Remove code related to trailer recording

Unlikely to be used and can be replicated with mods in a much better
way.

dengr1065 committed:
Do not reuse canvas between savegame loads

Remove lastCanvas/lastContext cache for reusing canvas as creating a new
canvas does not cause performance issue. Additionaly, remove smoothing
classes and buffer registration as they are not used anywhere. Finally,
remove redundant stageDestroyed calls in the InGameState.

dengr1065 committed:
Remove and simplify SCSS styles

Remove some unused classes, mixins and variables, then inline ones that
have just a few usages. Also remove a few redundant properties and
replace align-items: center + justify-items: center with place-items:
center.

dengr1065 committed:
Replace --ui-scale variable with rem unit

Replace --ui-scale inline style on the <html> element with font-size set
to UI scale * 10px and modify SCSS D(...) mixin to use the rem unit
instead of multiplying by var(--ui-scale). This makes it possible to get
rid of mixin hell and makes following UI scale in external stylesheets
(mods) much easier.

dengr1065 committed:
Replace usages of D() and S() SCSS mixins

Replace all instances of D() and S() mixins with rem units. Also,
replace deprecated slash division with math.div as suggested by
dart-sass. Finally, remove the mixins as they are no longer used.

dengr1065 committed:
Re-implement error handler

shapez used to have a nice error handler, but later it was removed. To
help debug game in both development and release cases, add a simple
error handler/screen that displays the error stack trace, lists the
installed mods and shows some build information. It also allows copying
this information to the system clipboard.

Having such an error screen should make mod development easier once the
local mod error handling is removed.

dengr1065 committed:
Simplify pointless try/catch wrappers

Remove try/catch wrappers that just log and/or rethrow the error, use
the cause chain for rethrows with descriptions. Remove inner try/catch
for mod loading to make errors more obvious.

dengr1065 committed:
Update webpack and simplify config

Update some webpack dependencies and remove unused ones. Replace
worker-loader with built-in worker loading functionality and remove
default/implied options from the webpack configuration files.

codeman4033 committed:
Handle directories in mod protocol handler (#87)

* Handle directories in mod protocol handler
Previously we would return an error about a file not existing, which kinda makes sense, considering a directory isn't a file.
Now with this commit, if the url resolve to a directory, it will return a json array with the relative paths of the file in that directory.

* Make directory index work with ASAR archives

Also fix directory index having higher priority than normal requests and
simplify URL handling.

---------

Co-authored-by: Даниїл Григор'єв <dengr1065@gmail.com>

EmeraldBlock committed:
fix incorrect sprite size and position when using pixelAligned

dengr1065 committed:
Update Electron to 37.x

Update electron package from 31.3.0 to 37.2.3, with Chromium 138 and
Node.js 22.17. Also follow the electron-packager -> @electron/packager
rename and upgrade the package.

dengr1065 committed:
Remove unused logging code

Remove functions in logging.js that are no longer used. As this action
makes the circular-json npm package unused, drop it as well.

dengr1065 committed:
Introduce new Logger implementation

Remove old Logger class and replace it with a new implementation that
keeps the location of the log calls (for display in devtools). Remove
internals of the old implementation and their one external usage.

dengr1065 committed:
Minor CSS changes

Remove PostCSS plugins that are no longer needed, fix some styles and
partially remove the need to use a .keybinding class on kbd elements.

dengr1065 committed:
Split KEYCODES from KeyActionMapper

Move KEYCODES and related functions to a separate file. No further
refactoring is done yet.

dengr1065 committed:
Always detach HUDSettingsMenu input receiver

Add a cleanup override for HUDSettingsMenu that detaches its input
receiver in case it is attached (i.e. when using the menu buttons to
open settings or return to main menu).

ninja-spotted committed:
Merge branch 'master' into patch-1

ninja-spotted committed:
Correction to pt-PT translation

ninja-spotted committed:
Update base-pt-PT.yaml
